src/main/java/apoc/util/Util.java

```diff
@@ -700,4 +706,99 @@ public static boolean classExists(String className) {
         Object value = ApocConfiguration.get(loadType).get(key);
         return Optional.ofNullable(value).map(Object::toString);
     }
+
+    public static Stream<MapResult> runWithRetry( GraphDatabaseService db, Log log, String statement, long retries, List<String> additionalRetryCodes, Map<String,Object> params )
+    {
+        String transactionFailureMessage = "";
+        Long timesTried = 0L;
+        do
+        {
+
+            FutureTask<Result> resultFuture = new FutureTask<>( () -> {
+                try ( Transaction tx = db.beginTx() )
+                {
+                    Result result = ((Function<Map<String,Object>, Result>) (Map<String,Object> p) -> db.execute(withParamMapping(statement, p.keySet()), p)).apply( params );
+                    tx.success();
+                    return result;
+                }
+
+            });
+
+            try
+            {
+                POOL.submit( resultFuture );
+
+                // Query has not finished yet, wait.
+                while ( !resultFuture.isDone() )
+                {
+                    LockSupport.parkNanos( 100 );
+                }
```

Comment:
Does this wait time work? Or do we want to scrap the waiting entirely and try to do something by streaming the result? I'm just not sure how I'd go about doing that.